nlp/2021_07_07_neural_keyword_extraction/neural_keyword_extraction.ipynb

```diff
@@ -0,0 +1,1373 @@
+    "\n",
+    "But what if the required performance is not met by these *classical* methods?\n",
+    "\n",
+    "## The Future of Keyword Extraction\n",
+    "\n",
+    "Newly-developed auspicious extraction methods fall into the category of **neural keyword extraction**.\n",
+    "\n",
+    "The methods often utilise **sequence-to-sequence models** based on recurrent neural networks (RNNs) or Long Short-Term Memory (LSTM). Their objective is to transform a sequence of input words (the given document) into an abstract intermediate representation and generate a sequence of keywords from it.\n",
+    "\n",
+    "These models do not use words or phrases directly, which enables them to generate unseen keywords as well. This is called **keyword generation** and combines abstractive as well as extractive keywords.\n",
+    "\n",
+    "They report **impressive performance** increases over the classical extraction methods. However, training such models **requires a large collection of documents** and annotated keywords, since the final training step is usually supervised.\n",
+    "\n",
+    "In addition, many of the models' repositories are not well maintained which makes it **difficult to train** them, especially on different languages or domains. \n",
+    "\n",
+    "This leaves the quesition of how neural keyword extraction can already be used today.\n",
+    "\n",
+    "Follow the below sections in this notebook to learn how to use **two approaches in the neural keyword extraction category** and how they compare to classical extraction methods. 👇"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "Ak0IG15fsUVV"
+   },
+   "source": [
+    "## 🏗 Getting started: Install packages & download models\n",
+    "\n",
+    "The below cells will setup everything that is required to get started with keyword extraction:\n",
+    "\n",
+    "* Install packages\n",
+    "* Download additional resources"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "qbMg6Xyj8Xww",
+    "outputId": "3b0b8599-77d5-41bf-b295-0ca49733b28e"
+   },
+   "outputs": [],
+   "source": [
+    "!pip install --quiet git+https://github.com/boudinfl/pke.git\n",
+    "!pip install --quiet transformers\n",
+    "!pip install --quiet keybert\n",
+    "\n",
+    "# Download additional resources\n",
+    "!python -m nltk.downloader stopwords\n",
+    "!python -m nltk.downloader universal_tagset\n",
+    "!python -m spacy download en # Download English model"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "vEu54fBNsUPR"
+   },
+   "source": [
+    "## KeyBERT\n",
+    "\n",
+    "Strictly speaking [KeyBERT](https://github.com/MaartenGr/KeyBERT) is not an end-to-end neural keyword extraction model. \n",
```

Comment:
I was a bit surprised to see that you originally say that the NKE methods are based on RNNs/LSTMs and then the first two models are Transformers ones 😛 Would it be worth mentioning that some use transformers/attention mechanism architectures?